empiar/protocols/protocol_downloader.py

```diff
@@ -1,422 +1,277 @@
+    fmt = DataFormat  # alias
+    dataFormatsToExtensions = {fmt.MRC.name: [".mrc"],
+                               fmt.MRCS.name: [".mrcs"],
+                               fmt.TIFF.name: [".tif", ".tiff"],
+                               fmt.DM4.name: [".dm4"],
+                               fmt.HDF5.name: [".hdf"],
+                               fmt.EER.name: [".eer, .gain"]}
 
     def __init__(self, **args):
         EMProtocol.__init__(self, **args)
-        self.stepsExecutionMode = STEPS_PARALLEL  # Defining that the protocol contain parallel steps
+        self.stepsExecutionMode = STEPS_PARALLEL
 
+    # -------------------------- DEFINE param functions -----------------------
     def _defineParams(self, form):
-        # add a section
         form.addSection("Entry")
 
-        # add a parameter to capture the EMPIAR entry ID:
-        # name --> entryId, String param, default value 10200, you choose the label
-        # Ideally we want it in bold is "important", it should not be empty, and fill the help.
-        form.addParam("entryId", params.StringParam, label="EMPIAR identifier", default="10200",
-                      help="EMPIAR's entry identifier", important=True)
-        # add another parameter to set a limit of downloaded files:
-        # name-->amountOfImages, Integer param , default to 1, choose the label and the help
-        # it has to be positive (use "validators" argument, it expects a list of
-        # pyworkflow.protocol.params.Validator, look for the Positive Validator)
-        form.addParam("amountOfImages", params.IntParam, label="Number of files", default=1,
-                      help="Number of files to download", validators=[params.Positive])
-
-        # Download folder
-        form.addParam("downloadFolder", params.FolderParam, label="Download folder", important=True,
-                      help="Local folder to store downloaded files")
-
-        form.addParam("makeEntryFolder", params.BooleanParam, label="Make an entry folder", default=True,
-                      help="If activated it will create a subfolder under the 'Download folder' with the EMPIAR##### name")
-
-        form.addParam("gainUrl", params.StringParam, label="Gain url", help="Url in the ftp site where the gain image is")
+        form.addParam("entryId", params.StringParam,
+                      label="EMPIAR identifier",
+                      default="10200",
+                      help="EMPIAR's entry identifier",
+                      important=True)
 
+        form.addParam("downloadFolder", params.FolderParam,
+                      label="Download folder", important=True,
+                      help="Local folder to store downloaded files")
 
+        form.addParam("amountOfImages", params.IntParam,
+                      label="Number of files", default=1,
+                      help="Number of files to download")
+
+        form.addParam("fileExt", params.EnumParam,
```

Comment:
How about having a text field instead of an enum to be more flexible?

like. ".mrcs .XML" and download only those file matching the patterns?

All have worked fine and very likely with current options all is covered.

Reply:
Trusting user input is the worst thing 😁

Reply:
The idea of enum here is to match with the data formats allowed by empiar. there's a separate option for xml/mdoc

Reply:
@pconesa I gave it another thought and you are right. If a user wants to download both tif and ever that wouldn't be possible. I'll make a string Param with a list of extensions